cmd/gpu_plugin/gpu_plugin.go

```diff
@@ -651,6 +794,34 @@ func main() {
 
 	plugin := newDevicePlugin(prefix+sysfsDrmDirectory, prefix+devfsDriDirectory, opts)
 
+	if plugin.options.wslScan {
+		klog.Info("WSL mode requested")
+
+		if plugin.options.resourceManagement {
+			klog.Error("Resource management is not supported within WSL. Please disable resource management.")
+
+			os.Exit(1)
+		}
+
+		if plugin.options.enableMonitoring {
+			klog.Error("Monitoring is not supported within WSL. Please disable monitoring.")
+
+			os.Exit(1)
+		}
+
+		if plugin.options.levelzero {
+			klog.Error("WSL requires Level-Zero service to operate. See README for help.")
```

Comment:
Shouldn't this be rather:
```suggestion
		if !plugin.options.levelzero {
			klog.Error("WSL requires Level-Zero service to operate. See README for help.")
```

But I wonder why L0 use is not enabled automatically if WSL needs it?

Reply:
Yes, this was badly converted from the previous "enable levelzero with socket path" method. And yes, if WSL is enabled, levelzero should be enabled at the same time.